### PR TITLE
Rename Attachments Integration tests

### PR DESCRIPTION
**Pull request by deepikaSingh2711**
## Describe your changes

This PR introduces integration tests to cover critical edge cases in the attachment renaming functionality. The implemented scenarios include **renaming a file with unsupported characters** (e.g., /), **renaming multiple attachments where one of the filenames contains invalid characters**, and attempting to **rename attachments using a user who lacks the necessary SDM roles.** 

To support the validation of unsupported characters, a minor enhancement was made in **API.java** to generically handle the sap-messages header from the draft response. This change led to the **refactoring of a few existing test cases to ensure consistency with the new logic**. All refactored tests have been **verified** and are **functioning correctly.**

The scenario involving **renaming an attachment to a name that already exists** within the entity was not implemented in this PR, as it is already covered by existing test cases. 

A new test case was added to validate rename behavior when performed by a user without **SDM roles**. But execution of this test is currently pending due to the unavailability of valid credentials.
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [x] I follow [Java Development Guidelines for SAP](https://help.sap.com/docs/SAP_COMMERCE/531a7d44feed44f99866946a4958b679/2e2d35a17d0e4ab8a0282d660d2531c1.html?locale=en-US&version=2205)
- [x] I have tested the functionality on my cloud environment.
- [x] I have  provided sufficient automated/ unit tests for the code.
- [x] I have increased or maintained the test coverage.
- [x] I have ran integration tests on my cloud environment.
- [x] I have validated blackduck portal for any vulnerability after my commit.

## Upload Screenshots/lists of the scenarios tested
- [ ] I have Uploaded Screenshots or added lists of the scenarios tested in description

### CODE REVIEW

File: sdm/src/test/java/integration/com/sap/cds/sdm/Api.java [hunk 1]
```diff
@@ -140,6 +140,10 @@ public String saveEntityDraft(
             System.out.println("Save entity failed. Error : " + draftResponseBodyString);
             return (draftResponseBodyString);
           }
+          String sapMessages = draftResponse.header("sap-messages");
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/Api.java:142]: Why is this needed?

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 2]
```diff
@@ -538,9 +570,131 @@ void testRenameSingleDuplicate() {
   }
 
   @Test
-  @Order(11)
+  @Order(12)
+  void testRenameMultipleEntitiesWithOneUnsupportedCharacter() {
+    System.out.println(
+        "Test (12) : rename multiple files out of which one file name contains unsupported characters");
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: NIT : Please keep it as Rename not rename

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 3]
```diff
@@ -538,9 +570,131 @@ void testRenameSingleDuplicate() {
...
+            && response.contains("note/invalid")) {
+          testStatus = true;
+        }
+      } else {
+        api.saveEntityDraft(appUrl, serviceName, entityName, srvpath, entityID);
+      }
+    }
+
+    if (!testStatus) {
+      fail(" Attachment was renamed which one file name contains unsupported characters");
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: NIT : Remove the extra space in between " and Attachment

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 4]
```diff
@@ -538,9 +570,131 @@ void testRenameSingleDuplicate() {
...
+        testStatus = false;
+        System.out.println("Entity was not put into draft mode: " + apiResponse);
+      }
+    } catch (Exception e) {
+      testStatus = false;
+      System.out.println("Exception during renaming entities: " + e.getMessage());
+    }
+
+    if (!testStatus) {
+      fail("Attachment got renamed without SDM role, which is not expected.");
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: You can change the message to -> "Attachment got renamed without SDM roles"

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 5]
```diff
@@ -457,8 +457,40 @@ void testRenameEntities() {
...
+        if (response.contains("Rename unsuccessful")
+            && response.contains("unsupported characters")) {
+          testStatus = true;
+        }
+      } else {
+        api.saveEntityDraft(appUrl, serviceName, entityName, srvpath, entityID);
+      }
+    }
+    if (!testStatus) {
+      fail("Attachment was renamed with unsupported characters or error was not detected");
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Change the message to -> "Attachment was renamed with unsupported characters"

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 6]
```diff
@@ -538,9 +570,131 @@ void testRenameSingleDuplicate() {
...
+    if (response.code() != 200) {
+      System.out.println("Token generation failed. Response code: " + response.code());
+      String errorBody = response.body().string();
+      System.out.println("Error body: " + errorBody);
+    }
+    token = new ObjectMapper().readTree(response.body().string()).get("access_token").asText();
+    response.close();
+    Map<String, String> config = new HashMap<>();
+    config.put("Authorization", "Bearer " + token);
+    api = new Api(config);
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Can all this be done in the setup method itself? Including token generation and instantiating api.java

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 7]
```diff
@@ -538,9 +570,131 @@ void testRenameSingleDuplicate() {
...
+                    + "/oauth/token?grant_type=password&username="
+                    + username2
+                    + "&password="
+                    + password2)
+            .method("POST", body)
+            .addHeader("Authorization", basicAuth)
+            .build();
+    Response response = client.newCall(request).execute();
+    if (response.code() != 200) {
+      System.out.println("Token generation failed. Response code: " + response.code());
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java:643]: Why is this needed?

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 8]
```diff
@@ -538,9 +570,131 @@ void testRenameSingleDuplicate() {
...
+      String errorBody = response.body().string();
+      System.out.println("Error body: " + errorBody);
+    }
+    token = new ObjectMapper().readTree(response.body().string()).get("access_token").asText();
+    response.close();
+    Map<String, String> config = new HashMap<>();
+    config.put("Authorization", "Bearer " + token);
+    api = new Api(config);
+    System.out.println("Test (8) : Rename single attachment, reference, and footnote");
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Why is this needed?

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 9]
```diff
@@ -457,8 +457,40 @@ void testRenameEntities() {
...
+        response =
+            api.renameAttachment(
+                appUrl, serviceName, entityName, facet[i], entityID, ID3[i], name[i]);
+        if (response.equals("Renamed")) counter++;
+      }
+      if (counter >= 2) {
+        counter = -1; // Reset counter for the next check
+        response = api.saveEntityDraft(appUrl, serviceName, entityName, srvpath, entityID);
+        System.out.println(response);
+        if (response.contains("Rename unsuccessful")
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Please compare the whole message, and not just parts of it

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 10]
```diff
@@ -538,9 +570,131 @@ void testRenameSingleDuplicate() {
...
+            api.renameAttachment(
+                appUrl, serviceName, entityName, facet[i], entityID, ID3[i], names[i]);
+        if (response.equals("Renamed")) successCount++;
+      }
+
+      if (successCount >= 2) {
+        response = api.saveEntityDraft(appUrl, serviceName, entityName, srvpath, entityID);
+        System.out.println("Save response: " + response);
+
+        if (response.contains("Rename unsuccessful")
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Please compare the whole message, and not just parts of it

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 11]
```diff
@@ -538,9 +570,131 @@ void testRenameSingleDuplicate() {
...
+              api.renameAttachment(
+                  appUrl, serviceName, entityName, facet[i], entityID, ID[i], name[i]);
+          if (!"Renamed".equals(apiResponse)) {
+            testStatus = false;
+            System.out.println(facet[i] + " was not renamed: " + apiResponse);
+          }
+        }
+        if (testStatus) {
+          apiResponse = api.saveEntityDraft(appUrl, serviceName, entityName, srvpath, entityID);
+          if (!apiResponse.contains(
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Please compare the whole message, and not just parts of it

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 12]
```diff
@@ -538,9 +570,131 @@ void testRenameSingleDuplicate() {
...
+              "You do not have the required permissions to update attachments")) {
+            testStatus = false;
+            System.out.println("Entity draft not saved: " + apiResponse);
+          }
+        } else {
+          api.saveEntityDraft(appUrl, serviceName, entityName, srvpath, entityID);
+        }
+      } else {
+        testStatus = false;
+        System.out.println("Entity was not put into draft mode: " + apiResponse);
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: No need to add an else condition for each call, refer to the other test cases

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 13]
```diff
@@ -698,15 +852,16 @@ void testUpdateValidSecondaryProperty_beforeEntityIsSaved_single() throws IOExce
       }
       if (counter >= 2)
         response = api.saveEntityDraft(appUrl, serviceName, entityName, srvpath, entityID3);
-      if (response.equals("Saved")) testStatus = true;
+      if (response.contains("The following secondary properties are not supported"))
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Please compare the whole message, and not just parts of it

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 14]
```diff
@@ -774,7 +929,7 @@ void testUpdateValidSecondaryProperty_afterEntityIsSaved_single() {
       }
       if (counter >= 2)
         response = api.saveEntityDraft(appUrl, serviceName, entityName, srvpath, entityID3);
-      if (response.equals("Saved")) {
+      if (response.contains("The following secondary properties are not supported")) {
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Please compare the whole message, and not just parts of it

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 15]
```diff
@@ -1329,7 +1484,7 @@ void testUpdateValidSecondaryProperty_afterEntityIsSaved_multipleAttachments() {
           && Updated3[1]
           && Updated3[2]) {
         response = api.saveEntityDraft(appUrl, serviceName, entityName, srvpath, entityID3);
-        if ("Saved".equals(response)) {
+        if (response.contains("The following secondary properties are not supported")) {
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Please compare the whole message, and not just parts of it, same for couple more occurrences above

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_SingleFacet.java [hunk 16]
```diff
@@ -467,9 +495,113 @@ void testRenameSingleAttachmentDuplicate() {
...
+    if (response.code() != 200) {
+      System.out.println("Token generation failed. Response code: " + response.code());
+      String errorBody = response.body().string();
+      System.out.println("Error body: " + errorBody);
+    }
+    token = new ObjectMapper().readTree(response.body().string()).get("access_token").asText();
+    response.close();
+    Map<String, String> config = new HashMap<>();
+    config.put("Authorization", "Bearer " + token);
+    api = new Api(config);
```
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_SingleFacet.java]: Can we add this token code to setup method

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_SingleFacet.java
Comment (rishikunnath2747) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_SingleFacet.java:1]: Please compare the whole message, and not just parts of it for all occurences in this file

**Review by rishikunnath2747 — CHANGES_REQUESTED** [review]
Please make the requested changes

File: sdm/src/test/java/integration/com/sap/cds/sdm/Api.java (hunk 1 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/Api.java:142]: There are some warning messages included in the sap-messages header along with the 200 response. We need to extract this header and validate the message content. Additionally, I have refactored the other associated test cases that were impacted by this change

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 2 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Done

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 3 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Done

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 4 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Done

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 5 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Done

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 6 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Moved to setup method

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 8 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: removed, updated accordingly

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 9 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: updated Accordingly with full message

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 10 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: updated Accordingly with full message

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 11 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: updated Accordingly with full message

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 12 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: Sure, Removed it

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 13 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: updated Accordingly with full message

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 14 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: updated Accordingly with full message

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 15 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java]: updated Accordingly with full message

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_SingleFacet.java (hunk 16 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_SingleFacet.java]: added in setup method

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_SingleFacet.java
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_SingleFacet.java:1]: updated Accordingly with full message

File: sdm/src/test/java/integration/com/sap/cds/sdm/Api.java (hunk 1 above)
Comment (rashmiangadi11) [reply sdm/src/test/java/integration/com/sap/cds/sdm/Api.java:142]: Can u remove System.out and replace with log.error or info

File: sdm/src/test/java/integration/com/sap/cds/sdm/Api.java [hunk 17]
```diff
@@ -139,6 +139,10 @@ public String saveEntityDraft(
...
+          if (sapMessages != null && !sapMessages.isEmpty()) {
+            return sapMessages;
+          }
           return "Saved";
         } catch (IOException e) {
           System.out.println("Could not save entity : " + e);
```
Comment (rashmiangadi11) [inline sdm/src/test/java/integration/com/sap/cds/sdm/Api.java:148]: Remove here as well

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java [hunk 18]
```diff
@@ -89,17 +95,41 @@ static void setup() throws IOException {
...
+                    + "&password="
+                    + password2)
+            .method("POST", body)
+            .addHeader("Authorization", basicAuth)
+            .build();
+
     Response response = client.newCall(request).execute();
+    Response responseNoRoles = client.newCall(requestNoRoles).execute();
     if (response.code() != 200) {
       System.out.println("Token generation failed. Response code: " + response.code());
```
Comment (rashmiangadi11) [inline sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java:113]: Remove all system out's

**Review by rishikunnath2747 — APPROVED** [review]

**Review by rishikunnath2747 — CHANGES_REQUESTED** [review]
Make sure any job related changes are done before merging

**Comment by deepikaSingh2711** [comment]
> Make sure any job related changes are done before merging

added the credentials to secret

File: sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java (hunk 18 above)
Comment (deepikaSingh2711) [reply sdm/src/test/java/integration/com/sap/cds/sdm/IntegrationTest_MultipleFacet.java:113]: System.out removals excluded as discussed

**Review by rishikunnath2747 — APPROVED** [review]
LGTM

**Review by rashmiangadi11 — APPROVED** [review]
